doc/p2p-bad-ports.md

```diff
@@ -0,0 +1,109 @@
+When Bitcoin Core automatically opens outgoing P2P connections it chooses
+a peer (address and port) from its list of potential peers. This list is
+populated with unchecked data, gossiped over the P2P network by other peers.
+
+It may happen that a malicious actor gossips an address:port where no Bitcoin
+node is listening. Or even where another service is listening which is not
+related to the Bitcoin network. As a result, this service may occasionally
+get connection attempts from Bitcoin nodes.
+
+"Bad" ports are ones that are used by services which usually are not opened to
+the public and usually require authentication. A connection attempt (by Bitcoin
+Core, trying to connect because it thinks there is a Bitcoin node on that
+address:port) to such service may be considered a malicious action by an
+ultra-paranoid administrator. An example for such a port is 22 (ssh). On the
+other hand, connection attempts to public services that usually do not require
+authentication are unlikely to be considered a malicious action,
+e.g. port 80 (http).
+
+Following is a list of "bad" ports which Bitcoin Core avoids when choosing a
+peer to connect to. If a node is listening on such a port, it will likely get
+less incoming connections.
+
+1
+7
+9
+11
+13
+15
+17
+19
+20
+21
+22
+23
+25
+37
+42
+43
+53
+69
+77
+79
+87
+95
+101
+102
+103
+104
+109
+110
+111
+113
+115
+117
+119
+123
+135
+137
+139
+143
+161
+179
+389
+427
+465
+512
+513
+514
+515
+526
+530
+531
+532
+540
+548
+554
+556
+563
+587
+601
+636
+989
+990
+993
+995
+1719
+1720
+1723
+2049
+3659
+4045
+5060
+5061
+6000
+6566
+6665
+6666
+6667
+6668
+6669
+6697
+10080
+
+For further information see:
+[pull/23306](https://github.com/bitcoin/bitcoin/pull/23306#issuecomment-947516736)
+[pull/23542](https://github.com/bitcoin/bitcoin/pull/23542)
+[fetch.spec.whatwg.org](https://fetch.spec.whatwg.org/#port-blocking)
+[chromium.googlesource.com](https://chromium.googlesource.com/chromium/src.git/+/refs/heads/main/net/base/port_util.cc)
+[hg.mozilla.org](https://hg.mozilla.org/mozilla-central/file/tip/netwerk/base/nsIOService.cpp)
```

Comment:
Added 2 spaces after each link for line break:

```suggestion
For further information see:

[pull/23306](https://github.com/bitcoin/bitcoin/pull/23306#issuecomment-947516736)  
[pull/23542](https://github.com/bitcoin/bitcoin/pull/23542)  
[fetch.spec.whatwg.org](https://fetch.spec.whatwg.org/#port-blocking)  
[chromium.googlesource.com](https://chromium.googlesource.com/chromium/src.git/+/refs/heads/main/net/base/port_util.cc)  
[hg.mozilla.org](https://hg.mozilla.org/mozilla-central/file/tip/netwerk/base/nsIOService.cpp)
```

Reply:
My editor highlights extra whitespace but sees only a line break here and `test/lint/lint-whitespace.sh` seems happy.

Reply:
Added empty lines.